src/arbitrage.py

```diff
@@ -80,6 +83,48 @@ class API(object):
     def __init__(self, args=None):
         self.name = args
 
+    def get(self, stock):
+        exps = self.get_expiries(stock)
+        spot = self.get_spot_price(stock)
+        premiums = {}
+        for exp in exps:
+            premiums[exp] = self.get_option_premiums(stock, exp, spot)
+        return StockData(stock, exps, spot, premiums)
+
+    def get_expiries(self, stock):
+        if self.name == "tradier":
+            option_expiries = requests.get('https://sandbox.tradier.com/v1/markets/options/expirations',
+                                           params={'symbol': stock, 'includeAllRoots': 'true', 'strikes': 'false'},
+                                           headers={'Authorization': 'Bearer Bfo8MwBCA6lFOqWSdWIe1Ke7IigA',
+                                                    'Accept': 'application/json'})
+            expiries = option_expiries.json()
+            return expiries['expirations']['date']
+
+    def get_spot_price(self, stock):
+        if self.name == "tradier":
+            response = requests.get('https://sandbox.tradier.com/v1/markets/quotes',
+                                    params={'symbols': stock, 'greeks': 'false'},
+                                    headers={'Authorization': 'Bearer Bfo8MwBCA6lFOqWSdWIe1Ke7IigA',
+                                             'Accept': 'application/json'})
+            quote = response.json()
+            return quote["quotes"]['quote']["ask"]
+
+    def get_option_premiums(self, stock, expiry, stock_price):
+        if self.name == "tradier":
+            result = [0]*2
+            response = requests.get('https://sandbox.tradier.com/v1/markets/options/chains',
+                                    params={'symbol': stock, 'expiration': expiry, 'greeks': 'false'},
+                                    headers={'Authorization': 'Bearer Bfo8MwBCA6lFOqWSdWIe1Ke7IigA',
+                                             'Accept': 'application/json'})
+            options = response.json()['options']['option']
+            for op in options:
+                if op["strike"] == stock_price:
```

Comment:
This doesn't work. The strike will not exactly be at the stock price. You choose the strike closest to the current stock price.

Reply:
fixed